src/App.js

```diff
@@ -2,39 +2,50 @@ import React, { useState } from 'react';
 import { Route } from 'react-router-dom';
 import data from './data';
 
+
 // Components
 import Navigation from './components/Navigation';
 import Products from './components/Products';
 import ShoppingCart from './components/ShoppingCart';
+import { ProductContext} from './components/Contexts/ProductContext';
+import { CartContext } from './components/Contexts/CartContext'
 
 function App() {
 	const [products] = useState(data);
 	const [cart, setCart] = useState([]);
 
 	const addItem = item => {
 		// add the given item to the cart
+		if (cart) {
```

Comment:
Interesting. I'm assuming this is so that it adds a count if there's already an item in the cart array? How did you come to this logic?